src/cmd/name.rs

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+//! `stg name` implementation.
+
+use std::io::Write;
+
+use anyhow::{anyhow, Result};
+use clap::{Arg, ArgMatches};
+use termcolor::WriteColor;
+
+use crate::{
+    argset,
+    branchloc::BranchLocator,
+    ext::RepositoryExtended,
+    patch::SingleRevisionSpec,
+    stack::{InitializationPolicy, Stack, StackAccess, StackStateAccess},
+};
+
+pub(super) const STGIT_COMMAND: super::StGitCommand = super::StGitCommand {
+    name: "name",
+    category: super::CommandCategory::PatchInspection,
+    make,
+    run,
+};
+
+fn make() -> clap::Command {
+    clap::Command::new(STGIT_COMMAND.name)
+        .about("Print patch name of a StGit revision")
+        .long_about(
+            "Print the patch name of a StGit revision.\n\
+             \n\
+             Try to get the name of the patch in the current \
+             branch as specified by a StGit revision. Revisions \
+             can be specified in the all the forms accepted by \
+             \"stg id\" command.",
+        )
+        .arg(argset::branch_arg())
+        .arg(
+            Arg::new("qualified")
+                .long("qualified")
+                .short('q')
+                .help("Prefix the patch name with the branch name")
+                .action(clap::ArgAction::SetTrue),
+        )
+        .arg(
+            Arg::new("stgit-revision")
+                .value_name("revision")
+                .allow_hyphen_values(true)
+                .value_parser(clap::value_parser!(SingleRevisionSpec))
+                .help("StGit revision"),
+        )
+}
+
+fn run(matches: &ArgMatches) -> Result<()> {
+    let repo = gix::Repository::open()?;
+
+    let stack = Stack::from_branch_locator(
+        &repo,
+        matches.get_one::<BranchLocator>("branch"),
+        InitializationPolicy::AllowUninitialized,
+    )?;
+
+    let oid = matches
+        .get_one::<SingleRevisionSpec>("stgit-revision")
+        .map(|spec| spec.resolve_object(&repo, &stack).map(|object| object.id))
+        .transpose()?
+        .unwrap_or_else(|| stack.get_branch_head().id);
+
+    let Some(patch_name) = stack
+        .all_patches()
+        .find(|name| oid == stack.get_patch_commit_id(name))
+    else {
+        return Err(anyhow!("patch name not found for revision `{oid}`"));
+    };
```

Comment:
Done. Before `stg name` failed when called without arguments on a branch without patches. I think it makes sense to keep that error so I changed the initialization to `InitializationPolicy::RequireInitialized`. I think that makes more sense: if there is a stack but (currently) no patches you get `{base}` otherwise (on a normal branch or detached head) you get `stack not initialized`.